glusterd2/xlator/options/options.go

```diff
@@ -324,10 +324,84 @@ func ValidateStr(o *Option, val string) error {
 
 // ValidateTime validates if the option is valid time format
 func ValidateTime(o *Option, val string) error {
-	if validate.IsTime(val, "hh:mm:ss") != true {
-		return ErrInvalidArg
+	var time int
+	var t string
+	var err error
+	if strings.HasSuffix(val, "sec") {
+		value := strings.TrimSuffix(val, "sec")
+		err = ValidateRange(o, value)
```

Comment:
`ValidateRange` is already being called at the end of the if block.

Reply:
the ValidateRange called at the end of the if else block is for the input cases with input values that need conversion. The validate range takes a string as an input, If the input from the command is just a number it is considered as a string and passed over to ValidateRange. If it has a string like "s", "hr" and so on, these have to be trimmed off. And if the value is given in any format other than seconds, then it has to be converted. For conversation, we have to convert it from string to int and then to pass it to validarange we need to parse it back to string. that why all this confusion. 

Again the answer for this comment is, the ValidRange will be called for all the input value which needed a conversion. This includes all formats other than "50", "50s", and "50sec"